PuzzleTimeObjective-C/PuzzleTime/GameViewController.m

```diff
@@ -0,0 +1,382 @@
+//
+//  GameViewController.m
+//  PuzzleTime
+//
+//  Created by Wei Shan on 1/11/16.
+//  Copyright © 2016 Wei Shan. All rights reserved.
+//
+
+#import "GameViewController.h"
+#import "PuzzleCell.h"
+#import "UIView+Snapshot.h"
+
+@interface GameViewController ()
+
+@end
+
+@implementation GameViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    [_collectionView registerNib:[UINib nibWithNibName:@"PuzzleCell" bundle:nil] forCellWithReuseIdentifier:@"Cell"];
+    
+    [self rebuildGame];
+    [self saveData];
+    self.automaticallyAdjustsScrollViewInsets = NO;
+    
+    //Attach long press gesture to collectionView
+    UILongPressGestureRecognizer *longPressGR = [[UILongPressGestureRecognizer alloc]
+                                          initWithTarget:self action:@selector(handleLongPress:)];
+    longPressGR.delegate = self;
+    longPressGR.delaysTouchesBegan = YES;
+    [_collectionView addGestureRecognizer:longPressGR];
+    
+    //Attach swipe gesture
+    UISwipeGestureRecognizer *swipeLeftGR = [[UISwipeGestureRecognizer alloc] initWithTarget:self action:@selector(handleSwipeGesture:)];
+    swipeLeftGR.delegate = self;
+    swipeLeftGR.direction = UISwipeGestureRecognizerDirectionLeft;
+    UISwipeGestureRecognizer *swipeRightGR = [[UISwipeGestureRecognizer alloc] initWithTarget:self action:@selector(handleSwipeGesture:)];
+    swipeRightGR.delegate = self;
+    swipeRightGR.direction = UISwipeGestureRecognizerDirectionRight;
+    UISwipeGestureRecognizer *swipeUpGR = [[UISwipeGestureRecognizer alloc] initWithTarget:self action:@selector(handleSwipeGesture:)];
+    swipeUpGR.delegate = self;
+    swipeUpGR.direction = UISwipeGestureRecognizerDirectionUp;
+    UISwipeGestureRecognizer *swipeDownGR = [[UISwipeGestureRecognizer alloc] initWithTarget:self action:@selector(handleSwipeGesture:)];
+    swipeDownGR.delegate = self;
+    swipeDownGR.direction = UISwipeGestureRecognizerDirectionDown;
+    
+    [self.view addGestureRecognizer:swipeLeftGR];
+    [self.view addGestureRecognizer:swipeRightGR];
+    [self.view addGestureRecognizer:swipeUpGR];
+    [self.view addGestureRecognizer:swipeDownGR];
+    
+    
+    
+}
+
+//Swipe Gesture
+- (void) handleSwipeGesture: (UISwipeGestureRecognizer *) swipeGestureRecognizer {
+    
+    NSIndexPath *emptySpotIndex = [NSIndexPath indexPathForItem:[_curItemsArray indexOfObject:@""] inSection:0];
+    NSIndexPath *movedItem = nil;
+    BOOL movable = NO;
+    
+    if (swipeGestureRecognizer.direction == UISwipeGestureRecognizerDirectionLeft) {
+        NSLog(@"swipe left");
+        if ((emptySpotIndex.item + 1) % _numberPerRow != 0) {
+            movedItem = [NSIndexPath indexPathForItem:emptySpotIndex.item + 1 inSection:0];
+            movable = YES;
+        }
```

Comment:
There might be a way for some of this "can move" logic to also go inside a separate `Game` class.